sound/soc/sof/ipc4-topology.c

```diff
@@ -2413,9 +2415,15 @@ static int sof_ipc4_widget_setup(struct snd_sof_dev *sdev, struct snd_sof_widget
 
 		msg->extension &= ~SOF_IPC4_MOD_EXT_PPL_ID_MASK;
 		msg->extension |= SOF_IPC4_MOD_EXT_PPL_ID(pipe_widget->instance_id);
+
+		dev_dbg(sdev->dev, "Create widget %s ID %#x instance %d - pipe %d - core %d\n",
+			swidget->widget->name, module_id, swidget->instance_id,
+			swidget->pipeline_id, swidget->core);
+	} else {
+		dev_dbg(sdev->dev, "Create widget %s instance %d - pipe %d - core %d\n",
+			swidget->widget->name, swidget->instance_id, swidget->pipeline_id,
+			swidget->core);
 	}
-	dev_dbg(sdev->dev, "Create widget %s instance %d - pipe %d - core %d\n",
-		swidget->widget->name, swidget->instance_id, swidget->pipeline_id, swidget->core);
```

Comment:
You are fixing a simple issue with very complex code, bad.  Every widget (pipe widget or a real widget) has ipc message, simply add the primary message print in this dev_dbg is enough.

```
dev_dbg(sdev->dev, "Create widget %s instance %d - pipe %d - core %d - primary message: %#x\n",
	swidget->widget->name, swidget->instance_id, swidget->pipeline_id, swidget->core, msg->primary);
```

please use %#x at least  for the primary message, hex is much more readable.

Reply:
@aiChaoSONG, it feels out of place to have the IPC message's primary printed just so that some python code would be able to extract the module id from it while it also needs to make sure that it is not taking it from a non module primary value (the pipeline).
I would rather prefer clean and human readable messages.

Reply:
@btian1 @aiChaoSONG what you are looking for is the module id and instance Id for the widget. You don't need the IPC msg data for that. And however you're using this info in your python code, please keep in mind that.while the module ID is immutable, the instance ID for a widget varies at runtime

Reply:
> while the module ID is immutable, the instance ID for a widget varies at runtime

As long as instance ID in kernel log and mtrace are the same in a single run, we are good.

Reply:
> I would rather prefer clean and human readable messages.

@ujfalusi I was focused on the minimum change in the kernel, but the hex ipc message is printed in ipc rx/tx, I will agree with you, let make it readable.